src/SUnit-Core/TestSuite.class.st

```diff
@@ -127,19 +127,22 @@ TestSuite >> run [
 
 { #category : #running }
 TestSuite >> run: aResult [
-	CurrentExecutionEnvironment runTestsBy: [
-		self runUnmanaged: aResult ]
+	
+	self runTestsInOnwEnvironments: aResult
 ]
 
 { #category : #running }
-TestSuite >> runUnmanaged: aResult [
+TestSuite >> runTestsInOnwEnvironments: aResult [
 	self setUp.
 	[ self tests
-		do: [ :each | 
-			each run: aResult.
-			self announceTest: each.
-			self changed: each ] ]
-		ensure: [ self tearDown ]
+		do: [ :aTestCase | 
+			| testEnv |
+			testEnv := aTestCase executionEnvironmentClass new.
+			testEnv beActiveDuring: [
+				aTestCase run: aResult.
+				self announceTest: aTestCase.
+				self changed: aTestCase ] ] 
+	] ensure: [ self tearDown ]
```

Comment:
The single environment per suite was an optimization to avoid overhead for every test during a big test suite.
For example each environment organizes a watch dog process to control the test running time.

Reply:
I see, maybe we should check if all the tests in the suite have the same environment class and split them only if it is not the case...